tendrl/commons/flows/__init__.py

```diff
@@ -115,7 +115,10 @@ def _execute_atom(self, atom_fqdn):
             ns, atom_name = atom_fqdn.split(".atoms.")
             ns, obj_name = ns.split(".objects.")
 
-            runnable_atom = self._ns.get_atom(obj_name, atom_name)
+            if hasattr(self, "obj"):
+                runnable_atom = self.obj._ns.get_atom(obj_name, atom_name)
+            else:
+                runnable_atom = self._ns.get_atom(obj_name, atom_name)
```

Comment:
Even the gluster integration fix patch: https://github.com/Tendrl/gluster-integration/pull/165 was tested including this PR. Without this it was not working